src/components/encoding-pane/function-picker.tsx

```diff
@@ -2,13 +2,19 @@ import {ExpandedType} from 'compassql/build/src/query/expandedtype';
 import {isWildcard} from 'compassql/build/src/wildcard';
 import * as React from 'react';
 import * as CSSModules from 'react-css-modules';
+import {OneOfFilter, RangeFilter} from 'vega-lite/build/src/filter';
+import {TimeUnit} from 'vega-lite/build/src/timeunit';
 import {ShelfFieldDef, ShelfFunction} from '../../models/shelf';
 import * as styles from './function-picker.scss';
 
 export interface FunctionPickerProps {
   fieldDef: ShelfFieldDef;
 
-  onFunctionChange: (fn: ShelfFunction) => void;
+  filter?: RangeFilter | OneOfFilter;
```

Comment:
Instead of adding `filter` as another possible params, I wonder if we should just change `fieldDef: ShelfFieldDef`

to 
````
fieldDefParts: {
   aggregate: ...,
   bin: ...,
   timeUnit: ...,
   field: ...
} 
```

which will be reuseable for both ShelfFieldDef and Filter.